one/src/migrations.rs

```diff
@@ -322,17 +331,26 @@ impl FSBlockStore {
 
 async fn block_from_path(block_path: PathBuf) -> Result<Option<(Cid, Vec<u8>)>> {
     if !block_path.is_file() {
+        if block_path.is_absolute() {
+            warn!(
+                path = %block_path.display(),
+                "block file not found, skipping");
+        } else {
+            warn!(
+                path = %block_path.display(),
+                "block file not found at relative path, skipping");
+        }
```

Comment:
I believe the check at line 33 is for skipping directories, so we shouldn't be getting here for any files.

Reply:
@smrz2001 it does trigger when I "accidentally" pass relative paths that can't be found, but it might _also_ trigger on empty dirs I guess :thinking: 

![image](https://github.com/user-attachments/assets/24da95f1-4a18-40c7-918f-da2440ddcd64)

Reply:
Any ideas on how to reconcile those checks? The issue I want to cover is it silently skipping every single block if the paths aren't correct (`!block_path.is_file()`). 

It's used in 2 spots:
- `blocks_from_list`, this looks fine
- `blocks_from_dir`, used during sharded imports. Since we still return `Ok`, I think it'd just emit a warn for not being able to find a block at a directory path (which arguably is correct), if that's not avoided by the dir check:
    ![image](https://github.com/user-attachments/assets/e007ee7e-d583-4240-b0d8-abf2914f5b78)


Anyway, happy for suggestions on how to improve the fix!

Reply:
nit: since the only difference between these branches is the message data, you could capture that and only have a single warn included.. e.g. one of these

```rust
warn!(path = %block_path.display(),
    relative_path = %!block_path.is_absolute(),
    "block file not found, skipping");

warn!(path = %block_path.display(),
    "block file not found at {} path, skipping", if block_path.is_absolute() {"absolute"} else {"relative"} );
 
let msg = if block_path.is_absolute() {} else {};
 warn!(path = %block_path.display(), "{}", msg);
```